### ISSUE TITLE
Add multiple levels of public access for projects

### ISSUE THREAD

**Issue by mikkonie** [bihealth/sodar-core#1014]
In discussions today it was also suggested to amend the current public access feature for multiple levels.

- Access controlled
    * The default as it is now
- Fully public
    * Corresponding to `public_guest_access` in the current implementation
- Publically discoverable regarding specific content
    * **TBD:** Which content, or allow selecting what to show?
- **TBD:** More?

This will not be done together with #1011, but will be done later.

Comments are welcome.

**Comment by holtgrewe** [comment]
Maybe we need a second access level/role "cursory guest access". This would be between "project discovery" and "guest" and allow to see the project and give access to "some" data in the project. E.g.,

- SODAR: see sample sheet but not iRODS data
- VarFish: see cases and annotated variants but no variant filtration
- Kiosc: list containers but not see details or access them

For further examples that we don't need to realize at all:

- Digestiflow: list sequencer, kits, flow cells, ... but no details
- sodar-core/smallfiles - browse files but no file access (could be configurable but as we only use this for storing attachments in digestiflow

So conceptually we have the "access levels":

- owner
- delegate
- contributor
- (maybe) "cursory guest"
- project discovery

There are two ways to assign these access levels.

First, role assignment directly to the project or containing categories once full role inheritance is implemented.

Second, setting the "visibility" of a project to "anonymous is cursory guest" or "anonymous is project discovery".

**Comment by mikkonie** [comment]
Yes, we indeed talked about this and I didn't include a mention in the spec. Good you brought it up.

The way I see it there are two options:

- Do as you specified above and include the cursory guest role as a SODAR Core level role
- Implement role customization (#288) and let SODAR Core based sites define such roles themselves

I think both approaches have their pros and cons. The first approach would be way more simple to implement, but it's a single user level for all sites, where in some cases we might want to have a more fine grained control over features to show. Not sure if such cases exist though.

Allowing sites to define their own roles is more flexible, but will most likely require more work for implementation and testing and might be complex for developers as well.

Of course, we could go with adding the "cursory guest" here now and in the future implementing role customization where this could be further extended if needed.
